src/main/java/org/jabref/gui/importer/GrobidPreferenceDialogHelper.java

```diff
@@ -7,29 +7,27 @@
 /**
  * Metadata extraction from PDFs and plaintext works very well using Grobid, but we do not want to enable it by default
  * due to data privacy concerns.
- * To make users aware of the feature, we ask each time before querying Grobid, giving the option to opt-out.
+ * To make users aware of the feature, we ask before querying Grobid, saving the users' preference for the future.
  */
-public class GrobidOptInDialogHelper {
+public class GrobidPreferenceDialogHelper {
 
     /**
-     * If Grobid is not enabled but the user has not explicitly opted-out of Grobid, we ask for permission to send data
-     * to Grobid using a dialog and giving an opt-out option.
+     * If the user has not explicitly opted-in/out of Grobid, we ask for permission to send data to Grobid by using
+     * a dialog. The users' preference is saved.
      *
      * @param dialogService the DialogService to use
      * @return if the user enabled Grobid, either in the past or after being asked by the dialog.
      */
     public static boolean showAndWaitIfUserIsUndecided(DialogService dialogService, GrobidPreferences preferences) {
-        if (preferences.isGrobidEnabled()) {
-            return true;
+        if (preferences.isGrobidPreference()) {
```

Comment:
We should use another word. This is NOT a JabRef setting (AKA preference), but a kind of "to prefer something". Moreover, this is asked at the first time only. Thus, I would rename it to `isGrobidUseAsked`. This is to strengthen that this is a one-time thing (in contrast to being asked at every run).

Reply:
Thank you for the observation, this would certainly make the logic more clear. I have made the changes as required.